ballerina/client.bal

```diff
@@ -0,0 +1,130 @@
+        SecretId|constraint:Error validated = constraint:validate(secretId);
+        if validated is constraint:Error {
+            return error Error(string `Request validation failed: ${validated.message()}`);
+        }
+        return self.externDescribeSecret(validated);
+    }
+
+    isolated function externDescribeSecret(SecretId secretId) returns DescribeSecretResponse|Error =
+    @java:Method {
+        name: "describeSecret",
+        'class: "io.ballerina.lib.aws.secretmanager.NativeClientAdaptor"
+    } external;
+
+    # Retrieves the contents of the encrypted fields from the specified version of a secret.
+    # ```ballerina
+    # secretmanager:SecretValue secret = check secretmanager->getSecretValue(secretId = "<aws-secret-id>");
+    # ```
+    #
+    # + request - The request object containing the details to identify the secret
+    # + return - An `secretmanager:SecretValue` containing the content of the secret, or an 
+    # `secretmanager:Error` if the request validation or the operation failed
+    remote function getSecretValue(*GetSecretValueRequest request) returns SecretValue|Error {
+        GetSecretValueRequest|constraint:Error validated = constraint:validate(request);
+        if validated is constraint:Error {
+            return error Error(string `Request validation failed: ${validated.message()}`);
+        }
+        return self.externGetSecretValue(validated);
+    }
+
+    isolated function externGetSecretValue(*GetSecretValueRequest request) returns SecretValue|Error =
+    @java:Method {
+        name: "getSecretValue",
+        'class: "io.ballerina.lib.aws.secretmanager.NativeClientAdaptor"
+    } external;
+
+    # Retrieves the contents of the encrypted fields for up to 20 secrets.
+    # ```ballerina
+    # secretmanager:BatchGetSecretValueResponse secret = check secretmanager->batchGetSecretValue(
+    #    secretIds = ["<aws-secret-id>"]);
+    # ```
+    #
+    # + request - The filters or secret IDs used to identify the secrets to retrieve
+    # + return - An `secretmanager:BatchGetSecretValueResponse` containing the contents of the secrets, or an 
+    # `secretmanager:Error` if the request validation or the operation failed
+    remote function batchGetSecretValue(*BatchGetSecretValueRequest request) returns BatchGetSecretValueResponse|Error {
+        BatchGetSecretValueRequest|constraint:Error validated = constraint:validate(request);
+        if validated is constraint:Error {
+            return error Error(string `Request validation failed: ${validated.message()}`);
+        }
+        if request.filters is () {
+            if request.secretIds is () {
+                return error Error("Either `filters` or `secretIds` must be provided in the request");
+            }
+        } else {
+            if request.secretIds is SecretId[] {
+                return error Error("The request cannot contain both `filters` and `secretIds` simultaneously");
+            }
+        }
```

Comment:
```suggestion
        if request.filters is () && request.secretIds is () {
                return error Error("Either `filters` or `secretIds` must be provided in the request");
        }

        if request.secretIds is SecretId[] && request.filters is SecretValueFilter[] {
            return error Error("The request cannot contain both `filters` and `secretIds` simultaneously");
        }
```

Reply:
@daneshk Empty array validation is already handled by the constraints validation.